### PR TITLE
Store the arguments used to trace the exported program in itself.

### PR DESCRIPTION
**Pull request by qihqi**
Currently this argument is optional and only be present if the way to produce ExportedProgram is indeed via tracing.

In particular, ExportedProgram loaded from a serialized files will NOT have original_traced_args. This way it retains BC as old  serialized files are still loadable.

Fixes #ISSUE_NUMBER

### CODE REVIEW

**Comment by qihqi** [comment]
@pytorchbot label "topic: export"

cc. @zhxchen17 @gmagogsfm

**Comment by angelayi** [comment]
You don't want to serialize the original traced arguments?

**Comment by qihqi** [comment]
> You don't want to serialize the original traced arguments?

The current discussion is the following:
1. If i serialize it then I still need to make it optional (because files serialized before this PR will not  have it).
2. Where to store? two options: a) same file as state_dict; b) separate file. a) is kinda of wierd, because original_inputs is not state. and b) is BC-breaking because serialize now need to return a tuple of 3 things and deserialize take 3 args. 

Make it optional and not serialize makes it 1. BC, and whatever we decide later on for 2. we will be able to glide to it gracefully.

**Comment by angelayi** [comment]
Ah ok, in that case, it's fine to ignore BC issues right now as no one is fully depending on the saved files yet -- after the branch cut on Friday 8/25, we'll probably have to bump the [version number](https://github.com/pytorch/pytorch/blob/1e9b590df989337d809fa33edd7ffc6cb60e70ff/torch/_export/serde/schema.py#L10). Can you implement the serialization part too? I think saving it to a separate file is better?

**Comment by qihqi** [comment]
> Ah ok, in that case, it's fine to ignore BC issues right now as no one is fully depending on the saved files yet -- after the branch cut on Friday 8/25, we'll probably have to bump the [version number](https://github.com/pytorch/pytorch/blob/1e9b590df989337d809fa33edd7ffc6cb60e70ff/torch/_export/serde/schema.py#L10). Can you implement the serialization part too? I think saving it to a separate file is better?

sure, done.
